src/ripple/app/tx/impl/DeleteAccount.cpp

```diff
@@ -0,0 +1,337 @@
+    ReadView const& view,
+    STTx const& tx)
+{
+    // The fee required for AccountDelete is one owner reserve.  But the
+    // owner reserve is stored in drops.  We need to convert it to fee units.
+    Fees const& fees {view.fees()};
+    std::pair<bool, std::uint64_t> const mulDivResult {
+        mulDiv (fees.increment, fees.units, fees.base)};
+    if (mulDivResult.first)
+        return mulDivResult.second;
+
+    // If mulDiv returns false then overflow happened.  Punt by using the
+    // standard calculation.
+    return Transactor::calculateBaseFee (view, tx);
+}
+
+namespace 
+{
+// Define a function pointer type that can be used to delete ledger node types.
+using DeleterFuncPtr = TER(*)(Application& app, ApplyView& view,
+    AccountID const& account, uint256 const& delIndex,
+    std::shared_ptr<SLE> const& sleDel, beast::Journal j);
+
+// Local function definitions that provides signature compatibility.
+TER
+offerDelete (Application& app, ApplyView& view,
+    AccountID const& account, uint256 const& delIndex,
+    std::shared_ptr<SLE> const& sleDel, beast::Journal j)
+{
+    return offerDelete (view, sleDel, j);
+}
+
+TER
+removeSignersFromLedger (Application& app, ApplyView& view,
+    AccountID const& account, uint256 const& delIndex,
+    std::shared_ptr<SLE> const& sleDel, beast::Journal j)
+{
+    return SetSignerList::removeSignersFromLedger (app, view, account);
+}
+
+TER
+removeDepositPreauthFromLedger (Application& app, ApplyView& view,
+    AccountID const& account, uint256 const& delIndex,
+    std::shared_ptr<SLE> const& sleDel, beast::Journal j)
+{
+    return DepositPreauth::removeDepositPreauthFromLedger (
+        app, view, delIndex, j);
+}
+
+struct CanDelete
+{
+    LedgerEntryType type;
+    DeleterFuncPtr deleter;
+};
+
+static constexpr std::array<CanDelete, 3> deletables
+{{
+    // This list is walked in order, so list from most to least common.
+    { ltOFFER,           offerDelete },
+    { ltSIGNER_LIST,     removeSignersFromLedger },
+    // {ltTICKET, ????},
+    { ltDEPOSIT_PREAUTH, removeDepositPreauthFromLedger },
+}};
+
+} // namespace
+
+TER
+DeleteAccount::preclaim (PreclaimContext const& ctx)
+{
+    AccountID const account {ctx.tx[sfAccount]};
+    AccountID const dst {ctx.tx[sfDestination]};
+
+    auto sleDst = ctx.view.read(keylet::account(dst));
+
+    if (!sleDst)
+        return tecNO_DST;
+
+    if ((*sleDst)[sfFlags] & lsfRequireDestTag && !ctx.tx[~sfDestinationTag])
+        return tecDST_TAG_NEEDED;
+
+    // Check whether the destination account requires deposit authorization.
+    if (ctx.view.rules().enabled(featureDepositAuth) &&
+        (sleDst->getFlags() & lsfDepositAuth))
+    {
+        if (! ctx.view.exists (keylet::depositPreauth (dst, account)))
+            return tecNO_PERMISSION;
+    }
+
+    auto sleAccount = ctx.view.read (keylet::account(account));
+    assert(sleAccount);
+    if (! sleAccount)
+        return terNO_ACCOUNT;
```

Comment:
Right.  My goal for this particular pass was to find all places where a missing account root in the ledger might crash the application.  For that pass I decided that overkill would be okay.  The checks are low cost.  It was easier for me to make the pass by simply thinking locally, inside of a given function, rather than worrying about currently guaranteed code flow between functions.

I don't feel strongly that this check is essential -- it's not.  But the presence of the check reduces my cognitive load somewhat when I'm thinking about what happens if an account root goes missing in the ledger.